### PR TITLE
DB-7229: remove unused backup system tables and procedures

### PR DESCRIPTION
**Pull request by jyuanca**

### CODE REVIEW

File: db-engine/src/main/java/com/splicemachine/db/iapi/sql/dictionary/DataDictionary.java [hunk 1]
```diff
@@ -230,17 +230,15 @@
     int SYSPERMS_CATALOG_NUM=20;
     int SYSUSERS_CATALOG_NUM=21;
     int SYSBACKUP_CATALOG_NUM=22;
-    int SYSBACKUPFILESET_CATALOG_NUM=23;
```
Comment (MurrayBrown) [inline db-engine/src/main/java/com/splicemachine/db/iapi/sql/dictionary/DataDictionary.java]: is it better to remove these, or leave the integers in the schema as 'unused placeholders', to reduce reverse compatibility problems after an upgrade?

Comment (jyuanca) [reply db-engine/src/main/java/com/splicemachine/db/iapi/sql/dictionary/DataDictionary.java]: Good question. We never dropped system tables before. I'm not aware of any consequences if we modify existing table number. I've run upgrade tests, and it looks good to me.

Comment (MurrayBrown) [reply db-engine/src/main/java/com/splicemachine/db/iapi/sql/dictionary/DataDictionary.java]: My suggestion is to leave the unchanged numbers alone, i.e. remove '23 and 24', but leave the numbers 'the same' for all the others 25 through 33, so we'd end the list on:
  SYSSNAPSHOT_NUM=32; and SYSTOKENS_NUM=33;

In this way, we wont have a problem accessing any records written to a pre-existing database that has already stored a data value based on the integers 25 through 33.  

Either that, or I am not understanding the data dictionary correctly!

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

File: db-engine/src/main/java/com/splicemachine/db/iapi/sql/dictionary/DataDictionary.java (hunk 1 above)
Comment (MurrayBrown) [reply db-engine/src/main/java/com/splicemachine/db/iapi/sql/dictionary/DataDictionary.java]: Agreed, I think i'm happier with that sort of approach (just to be safe?).

**Comment by jyuanca** [comment]
jenkins please test branch @cdh5.12.2

**Comment by jyuanca** [comment]
Please also review code changes on ee branch

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/SpliceDataDictionary.java [hunk 2]
```diff
@@ -1163,4 +1115,31 @@ public void upgradeSysColumnsWithUseExtrapolationColumn(TransactionController tc
             SpliceLogUtils.info(LOG, "SYS.SYSCOLUMNS upgraded: added column: USEEXTRAPOLATION.");
         }
     }
+
+    public void removeUnusedBackupTables(TransactionController tc) throws StandardException {
+        SchemaDescriptor sd = getSystemSchemaDescriptor();
+        TableDescriptor td = getTableDescriptor("SYSBACKUPFILESET", sd, tc);
+        if (td != null) {
+            dropTableDescriptor(td, sd, tc);
```
Comment (yxia92) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/SpliceDataDictionary.java]: Is it enough to just drop the table descriptor if we want to remove the unused backup tables? what about the columns of these tables, and the conglomerate as well as index conglomerates on these tables? Looking at DropTableConstantOperation.executeConstantAction(), dropping a user table seems to have more things to do. Is it similar for dictionary table? Thanks.

Comment (jyuanca) [reply splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/SpliceDataDictionary.java]: Good catch!

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Comment by jyuanca** [comment]
jenkins please test branch @cdh5.12.2

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Pull request merged**